android/src/main/java/com/snowplowanalytics/react/tracker/RNSnowplowTrackerModule.java

```diff
@@ -120,34 +135,79 @@ public void trackSelfDescribingEvent(ReadableMap event, ReadableArray contexts)
     }
 
     @ReactMethod
-    public void trackStructuredEvent(String category, String action, String label,
-                                     String property, Float value, ReadableArray contexts) {
-        Structured trackerEvent = EventUtil.getStructuredEvent(category, action, label,
-                property, value, contexts);
+    public void trackStructuredEvent(ReadableMap details,
+                                     ReadableArray contexts,
+                                     Promise promise) {
+
+        if (!(details.hasKey("category") &&
+           details.hasKey("action") &&
+           details.hasKey("label") &&
+           details.hasKey("property") // 'value' key deliberately removed if null/undefined, so no check on that.
+        )) {
+         promise.reject("ERROR", "SnowplowTracker: trackStructuredEvent() method - missing parameter with no default found");
+        }
+
+        Structured trackerEvent = EventUtil.getStructuredEvent(
+          details.getString("category"),
+          details.getString("action"),
+          details.getString("label"),
+          details.getString("property"),
+          details.hasKey("value") ? details.getDouble("value") : null,
```

Comment:
If you check with `details.isNull("value") ? ....` maybe it's not needed to remove the "value" param from the args.
Just wondering!

Reply:
Good point. Missed that completely. 

Ian's input in slack has made me question whether the pattern should change anyway though - setting null defaults to work around limitations of methods seems to abstract the solution away from the problem. 

Fixing the util methods, and conditionally removing builder arguments may be the preferable approach... I'm mulling it over!